src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/InstructionList.cs

```diff
@@ -407,13 +407,10 @@ internal void SwitchToBoxed(int index, int instructionIndex)
         {
             var instruction = _instructions[instructionIndex] as IBoxableInstruction;
 
-            if (instruction != null)
+            Instruction newInstruction = instruction?.BoxIfIndexMatches(index);
```

Comment:
It may get optimized away, but logically this introduces another null check in the case where instruction was already null, right?